carrot-log4j-aws-sns/src/main/java/com/carrotgarden/log4j/aws/sns/Appender.java

```diff
@@ -203,8 +209,11 @@ public boolean hasAmazonClient() {
 
 	/** provide amazon login credentials from file */
 	protected boolean ensureCredentials() {
-
-        if (hasCredentials()) {
+		if (hasAwsCredentialsOverride()) {
+			amazonCredentials = new AWSCredentialsProviderChain(new InstanceProfileCredentialsProvider(),
+																new EnvironmentVariableCredentialsProvider(),
```

Comment:
Could just throw the 2 below in this chain after and be done with it (or before)...new Chain(Classpath, Properties, Instance)?